File: summer/WebContent/js/MS/internal/data/ListSourceConverter.js
/**
 * ListSourceConverter
 */

define(["dojo/_base/declare", "system/Type", "data/IValueConverter"], 
		function(declare, Type, IValueConverter){
	var ListSourceConverter = declare("ListSourceConverter", IValueConverter,{
		constructor:function(){
		},
		
//		public object 
		Convert:function(/*object*/ o, /*Type*/ type, /*object*/ parameter, /*CultureInfo*/ culture) 
        {
            /*IList*/var il = null;
            /*IListSource*/var ils = o instanceof IListSource ? o : null;
 
            if (ils != null)
            { 
                il = ils.GetList(); 
            }
 
            return il;
        },

//        public object 
		ConvertBack:function(/*object*/ o, /*Type*/ type, /*object*/ parameter, /*CultureInfo*/ culture) 
        {
            return null; 
        } 
	});
	
	Object.defineProperties(ListSourceConverter.prototype,{
		  
	});
	
	Object.defineProperties(ListSourceConverter,{
		  
	});
	
	ListSourceConverter.Type = new Type("ListSourceConverter", ListSourceConverter, [IValueConverter.Type]);
	return ListSourceConverter;
});

File: org.summer.view.widget/src/org/summer/view/widget/data/ListSourceConverter.java
package org.summer.view.widget.data;
/*internal*/ public class ListSourceConverter implements IValueConverter
{ 
    //-----------------------------------------------------
    // 
    //  Constructors 
    //
    //------------------------------------------------------ 


    //-----------------------------------------------------
    // 
    //  Interfaces (IValueConverter)
    // 
    //----------------------------------------------------- 

    public Object Convert(Object o, Type type, Object parameter, CultureInfo culture) 
    {
        IList il = null;
        IListSource ils = o as IListSource;

        if (ils != null)
        { 
            il = ils.GetList(); 
        }

        return il;
    }

    public Object ConvertBack(Object o, Type type, Object parameter, CultureInfo culture) 
    {
        return null; 
    } 
}